Source/AFError.swift

```diff
@@ -343,12 +406,22 @@ extension AFError {
         guard case .responseSerializationFailed(let reason) = self else { return nil }
         return reason.failedStringEncoding
     }
+
+    public var destination: URL? {
+        guard case .downloadedFileMoveFailed(_, let destination) = self else { return nil }
+        return destination
+    }
 }
 
 extension AFError.ParameterEncodingFailureReason {
     var underlyingError: Error? {
-        guard case .jsonEncodingFailed(let error) = self else { return nil }
-        return error
+        switch self {
+        case let .jsonEncodingFailed(error),
+             let .customEncodingFailed(error):
+            return error
+        default:
```

Comment:
This switch should be made exhaustive so we don't miss cases in the future.

Reply:
Agreed. I would even make all switches exhaustive here. I actually didn't notice the `underlyingError` utility because they werent.

Reply:
I removed default cases from all switch statements where a newly added case would need to be handled. It's a bit more verbose but much safer. Let me know what you think